### PR TITLE
Add missing docstrings and updated old ones

### PR DESCRIPTION
**Pull request by Pradhvan**
* Added missing docstrings to the updated docstring pattern.
* Updated missing docstrings to the updated docstring pattern.

Closes: https://github.com/scanapi/scanapi/issues/337

### CODE REVIEW

**Comment by Pradhvan** [comment]
Need a bit of help identifying what type of data is taken in and returned in `scanapi/config_loader.py` for now I have filled in with placeholders.

File: scanapi/reporter.py [hunk 1]
```diff
@@ -11,13 +14,26 @@
...
+        """Creates a Reporter instance object."""
         self.output_path = output_path or "scanapi-report.html"
         self.template = template
 
     def write(self, results):
-        """Part of the Reporter instance that is responsible for writing scanapi-report.html."""
+        """Method responsible for writing scanapi-report.html.
+
+        Args:
+            results [type]: Fill me!
```
Comment (Pradhvan) [inline scanapi/reporter.py:34]: @camilamaia `results` would be dict? I did some tests saw some fake results that were dict. Still, a bit confused so left a placeholder here too.

Comment (camilamaia) [reply scanapi/reporter.py:34]: @Pradhvan is a generator object containing results for each request. Each result contains the information for a request.

https://scanapi.dev/docs_v1/configuration/custom_report.html#results

If you [run ScanAPI locally](https://github.com/scanapi/scanapi/wiki/Run-ScanAPI-Locally) and add a break point to the write method, you can check it better:

image


```python
$ source ../examples/demo-api/.env
$ poetry run scanapi run ../examples/demo-api/scanapi.yaml -c ../examples/demo-api/scanapi.conf -o ../examples/demo-api/scanapi-report.html
Loading file ../examples/demo-api/scanapi.conf
Loading file ../examples/demo-api/scanapi.yaml
Writing documentation
> /Users/camilamaia/workspace/scanapi-org/scanapi/scanapi/reporter.py(26)write()
-> template_path = self.template if self.template else "report.html"
(Pdb) results
<generator object EndpointNode.run at 0x10d33ce60>
(Pdb) results_as_a_list = list(results)
Making request GET http://demo.scanapi.dev/api/v1/health/
Making request POST http://demo.scanapi.dev/api/v1/api-token-auth/
Making request GET http://demo.scanapi.dev/api/v1/snippets/
Making request POST http://demo.scanapi.dev/api/v1/snippets/
Making request GET http://demo.scanapi.dev/api/v1/snippets/70/
Making request PUT http://demo.scanapi.dev/api/v1/snippets/70/
Making request PATCH http://demo.scanapi.dev/api/v1/snippets/70/
Making request GET http://demo.scanapi.dev/api/v1/snippets/70/highlight/
Making request DELETE http://demo.scanapi.dev/api/v1/snippets/70/
Making request GET http://demo.scanapi.dev/api/v1/users/
Making request GET http://demo.scanapi.dev/api/v1/users/3/
(Pdb) results_as_a_list
[{'response': <Response [200]>, 'tests_results': [{'name': 'snippets-api::health::status_code_is_200', 'status': 'passed', 'failure': None, 'error': None}, {'name': 'snippets-api::health::bode_equals_ok', 'status': 'passed', 'failure': None, 'error': None}], 'no_failure': True}, {'response': <Response [200]>, 'tests_results': [{'name': 'snippets-api::get_token::status_code_is_200', 'status': 'passed', 'failure': None, 'error': None}], 'no_failure': True}, {'response': <Response [200]>, 'tests_results': [{'name': 'snippets-api::snippets::list_all::status_code_is_200', 'status': 'passed', 'failure': None, 'error': None}], 'no_failure': True}, {'response': <Response [201]>, 'tests_results': [{'name': 'snippets-api::snippets::create::status_code_is_201', 'status': 'passed', 'failure': None, 'error': None}], 'no_failure': True}, {'response': <Response [200]>, 'tests_results': [{'name': 'snippets-api::snippets::details::status_code_is_200', 'status': 'passed', 'failure': None, 'error': None}], 'no_failure': True}, {'response': <Response [200]>, 'tests_results': [{'name': 'snippets-api::snippets::update_with_put::status_code_is_200', 'status': 'passed', 'failure': None, 'error': None}], 'no_failure': True}, {'response': <Response [200]>, 'tests_results': [{'name': 'snippets-api::snippets::update_with_patch::status_code_is_200', 'status': 'passed', 'failure': None, 'error': None}], 'no_failure': True}, {'response': <Response [200]>, 'tests_results': [{'name': 'snippets-api::snippets::highlight::status_code_is_200', 'status': 'passed', 'failure': None, 'error': None}], 'no_failure': True}, {'response': <Response [204]>, 'tests_results': [{'name': 'snippets-api::snippets::delete::status_code_is_204', 'status': 'passed', 'failure': None, 'error': None}], 'no_failure': True}, {'response': <Response [200]>, 'tests_results': [{'name': 'snippets-api::users::list_all::status_code_is_200', 'status': 'passed', 'failure': None, 'error': None}], 'no_failure': True}, {'response': <Response [200]>, 'tests_results': [{'name': 'snippets-api::users::details::status_code_is_200', 'status': 'passed', 'failure': None, 'error': None}], 'no_failure': True}]
```

Comment (camilamaia) [reply scanapi/reporter.py:34]: Maybe `results [generator]: generator of dicts resulting of Request run().` What do you think @Pradhvan ?

Comment (Pradhvan) [reply scanapi/reporter.py:34]: Thanks for the help. Will update the PR. 😄

File: scanapi/config_loader.py [hunk 2]
```diff
@@ -28,7 +33,16 @@ def __init__(self, stream: IO) -> None:
 
 
 def construct_include(loader: Loader, node: yaml.Node) -> Any:
-    """Include file referenced at node."""
+    """Include file referenced at node.
+
+        Args:
+            loader [class.Loader]: fill me!
```
Comment (camilamaia) [inline scanapi/config_loader.py:39]: @Pradhvan are you going to leave the `fill me` for now and fill it later? Or are you going to fill it still in this PR? I was waiting for it to approve the PR. But we can leave it for later, if you prefer :)

Comment (Pradhvan) [reply scanapi/config_loader.py:39]: Hey! @camilamaia will finish this PR and update the `fill me` this weekend. Then we can merge this. 😄

**Pull request closed without merging**